### PR TITLE
Fix telecomms to work with multi-z

### LINKED ISSUE #47075: Multi-Z issues
- Making a hole in an upper floor that has a lower open segment (i.e a room) only makes generic space. This requires you to make upper-floor baseturfs openspace and means players can't do any real room projects that span more than one floor in-game.
- No visual indicator of lower structures in space (i.e seeing the 'ceiling' of the lower level's floor/wall).
- Moving up/down as a non-ghost doesn't work anywhere (space, openspace, in no-grav, ect).
- Telecomms doesn't work on different floors without additional setups.

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request

This fixes #47075 

Telecomms on stations had to add a relay to each z-level to get the radio to work.  This removes that requirement and should free up some mapping space.

## Why It's Good For The Game

Less janky mapping hacks.  Also this could be useful for multi-z ruins in the future.

## Changelog

:cl:
fix: Fix telecomms relays to work across multi-z
/:cl:

### CODE REVIEW

**Comment by timothymtorres** [comment]
I haven't had a chance to test if this works yet.  When I can confirm if it works, I will report back here.

File: code/game/machinery/telecomms/machines/relay.dm [hunk 1]
```diff
@@ -23,7 +23,12 @@
 	// Add our level and send it back
 	var/turf/T = get_turf(src)
 	if(can_send(signal) && T)
-		signal.levels |= T.z
+		// Relays send signals to all ZTRAIT_STATION z-levels
+		if(SSmapping.level_trait(T.z, ZTRAIT_STATION))
+			for(var/z in SSmapping.levels_by_trait(ZTRAIT_STATION))
```
Comment (SuperNovaa41) [inline code/game/machinery/telecomms/machines/relay.dm:28]: better var names please :) https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#use-descriptive-and-obvious-names

Comment (timothymtorres) [reply code/game/machinery/telecomms/machines/relay.dm:28]: Fixed with da31852580b0e3c4a52e00579d8780baf8629ed1

**Comment by Mothblocks** [comment]
Fairly elegant

**Pull request merged**